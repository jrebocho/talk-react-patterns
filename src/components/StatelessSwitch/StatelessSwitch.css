.box {
  width: 200px;
  height: 100px;
  line-height: 100px;
  text-align: center;
  font-size: 60px;
}

.box:hover {
  cursor: pointer;
}

.green {
  background-color: green;
  color: yellow;
}

.red {
  background-color: red;
  color: white;
}
