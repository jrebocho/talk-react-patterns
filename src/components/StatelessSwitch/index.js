import React from 'react'
import PropTypes from 'prop-types'
import './StatelessSwitch.css'

const StatelessSwitch = ({ on }) => {
  const boxColor = on ? 'green' : 'red'

  return <div className={`box ${boxColor}`}>{on ? 'ON' : 'OFF'}</div>
}

StatelessSwitch.propTypes = {
  on: PropTypes.bool
}

export default StatelessSwitch
