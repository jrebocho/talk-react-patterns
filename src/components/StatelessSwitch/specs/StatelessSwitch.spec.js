import React from 'react'
import { shallow } from 'enzyme'
import StatelessSwitch from '../../StatelessSwitch'

describe('<StatelessSwitch />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<StatelessSwitch />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is on', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<StatelessSwitch on={true} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
