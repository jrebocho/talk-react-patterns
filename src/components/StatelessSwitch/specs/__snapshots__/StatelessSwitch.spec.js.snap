// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatelessSwitch /> renders correctly 1`] = `
<div
  className="box red"
>
  OFF
</div>
`;

exports[`<StatelessSwitch /> when the component is on renders correctly 1`] = `
<div
  className="box green"
>
  ON
</div>
`;
